src/classes/BDI_Donations.cls

```diff
@@ -711,5 +773,136 @@ public class BDI_Donations {
             }
         }
     }
+
+    /*******************************************************************************************************
+    * @description helper method to return true if the record is an elevate oppt/payment and false if not.
+    * @param dataImportToReview the DI to determine if it is from Elevate
+    * @return result 
+    */
+    private Boolean isElevatePaymentDI(DataImport__c di) {
+        return di.Payment_Elevate_Id__c != null;
+    }
+
+    /*******************************************************************************************************
+    * @description returns a list of DI that are not Elevate Payments for use with matching
+    * @param dataImports The list to remove elevatePayments from.
+    * @return nonElevateDIs  list of non-Elevate payment DIs 
+    */
+    private DataImport__c[] filterOutElevatePaymentDIs (DataImport__c[] dis) {
+        DataImport__c[] nonElevateDIs = new DataImport__c[]{};
+        for (DataImport__c di : dis) {
+            if (isElevatePaymentDI(di)) {
+                continue;
+            } else {
+                nonElevateDIs.add(di);
+            }
+        }
+        return nonElevateDIs;
+    }
+
+    /*******************************************************************************************************
+    * @description Populates new and updates existing payments as needed for elevate related transactions.
+    * @param dataImportsById map of Data import records by their ids. 
+    */
+    private void processPaymentsForElevateDIs (DataImport__c[] dataImports) {
+        Map<Id, npe01__OppPayment__c> paymentsForUpsertByDiId = new Map<Id, npe01__OppPayment__c>();
```

Comment:
This `paymentsForUpsertByDiID` is not being used  right?

Reply:
@KietTruongSFDO Yeah it was a leftover, removed it.